src/libcore/tests/test_xml.py

```diff
@@ -298,3 +298,166 @@ def test20_upgrade_tree(variant_scalar_rgb):
                                <float name="intIOR" value="1.33"/>
                            </bsdf>
                        </scene>""")
+
+@fresolver_append_path
+def test21_xml_save_plugin(variant_scalar_rgb):
+    from mitsuba.core import xml
+    from mitsuba.python.dict_to_xml import WriteXML
+    import os
+    abs_path = os.path.abspath('resources/data/scenes/dict/dict.xml')
+    writer = WriteXML(abs_path)
+    scene_dict = {
+            "type": "sphere",
+            "center" : [0, 0, -10],
+            "radius" : 10.0,
+        }
+    writer.configure(scene_dict)
+    s1 = xml.load_dict({
+        'type': 'scene',
+        'sphere':{
+            "type": "sphere",
+            "center" : [0, 0, -10],
+            "radius" : 10.0,
+        }
+        })
+    s2 = xml.load_file(abs_path)
+    assert str(s1) == str(s2)
+
+@fresolver_append_path
+def test22_xml_missing_type(variant_scalar_rgb):
+    from mitsuba.python.dict_to_xml import WriteXML
+    import os
+    abs_path = os.path.abspath('resources/data/scenes/dict/dict.xml')
```

Comment:
Could we delete the temporary file once the test finished? Also I wonder if this approach would work when running tests in parallel? We should either have a different file per test, or have the ability to catch the output string (and later use `xml.load_string`).

Reply:
There is now one scene folder per test, removed at the end of the unit test